CRM/Financial/BAO/Payment.php

```diff
@@ -169,6 +169,8 @@ protected static function loadRelatedEntities($id) {
     $contactID = self::getPaymentContactID($contributionID);
     list($displayName, $email)  = CRM_Contact_BAO_Contact_Location::getEmailDetails($contactID);
     $entities['contact'] = ['id' => $contactID, 'display_name' => $displayName, 'email' => $email];
+    $contact = civicrm_api3('Contact', 'getsingle', ['id' => $contactID, 'return' => 'email_greeting']);
+    $entities['contact']['email_greeting'] = $contact['email_greeting_display'];
```

Comment:
It seems $contact is not used anywhere below so can we change it to:
```php
$entities['contact']['email_greeting'] = civicrm_api3('Contact', 'getvalue', ['id' => $contactID, 'return' => 'email_greeting_display']);
```

Reply:
@monishdeb nope we can't - if it's empty getvalue will throw an error